src/wp-includes/rest-api/class-wp-rest-server.php

```diff
@@ -230,6 +230,33 @@ protected function json_error( $code, $message, $status = null ) {
 		return wp_json_encode( $error );
 	}
 
+	/**
+	 * Gets the encoding options passed to {@see wp_json_encode}.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param \WP_REST_Request $request The current request object.
+	 *
+	 * @return int The JSON encode options.
+	 */
+	protected function get_json_encode_options( WP_REST_Request $request ) {
+		$options = defined( 'REST_JSON_ENCODE_OPTIONS' ) ? REST_JSON_ENCODE_OPTIONS : 0;
```

Comment:
@TimothyBJacobs What are your thoughts on the advantages of this over the `WP_DEBUG_REST_API` constant proposed in the original trac ticket?

If I am following this, to define that I want my site to JSON pretty print, I'd use this, correct?:

```php
define( 'REST_JSON_ENCODE_OPTIONS', JSON_PRETTY_PRINT );
```

I don't know that I have a strong opinion on this, and admit that `WP_DEBUG_REST_API` achieves a little developer-facing consistency but also possibly over-implies what you're getting in exchange for setting the constant. I'd almost expect that if I say "I'm debugging the REST API" that I'd start getting [telemetry](https://querymonitor.com/blog/2021/05/debugging-wordpress-rest-api-requests/) or something along with my request response, which would definitely not be the case here.

Naming things is hard. I'd love a second opinion :)

Reply:
> If I am following this, to define that I want my site to JSON pretty print, I'd use this, correct?:

Yep, that's correct!

Yeah, I think "debug" implies more features than we have, and isn't as clear that it is enabling "pretty print". But this let's us be very clear in what the constant supports and leaves us open to future expansion.